File: catkin_ws/src/learning_topic/src/person_subscriber.cpp
#include <ros/ros.h>
#include "learning_topic/Person.h"

// callback function
void personInfoCallback(const learning_topic::Person::ConstPtr& msg)
{
	// print
	ROS_INFO("Subscribe Person Info: name:%s  age:%d  sex:%d", msg->name.c_str(), msg->age, msg->sex);
}

int main(int argc, char **argv)
{
	// init ROS node
	ros::init(argc, argv, "person_subscriber");

	// create handler
	ros::NodeHandle n;

	// create Subscriber, topic /person_info, callback function poseCallback
	ros::Subscriber person_info_sub = n.subscribe("/person_info", 1, personInfoCallback);

	// loop wait
	ros::spin();

	return 0;
}

File: catkin_ws/src/learning_topic/scripts/person_subscriber.py
#!/usr/bin/env python
import rospy
from learning_topic.msg import Person

def personInfoCallback(msg):
    rospy.loginfo("Subscribe Person Info: name:%s  age:%d  sex:%d", msg.name, msg.age, msg.sex)

def person_subscriber():
    # ROS node init
    rospy.init_node('person_subscriber', anonymous=True)

    # create Subscriber, topic /person_subscriber, callback function poseCallback
    rospy.Subscriber("/person_info", Person, personInfoCallback)

    # wait
    rospy.spin()

if __name__ == '__main__':
    person_subscriber()